src/CameraGroupObs.cpp

```diff
@@ -32,11 +32,7 @@ void CameraGroupObs::insertObjectObservation(
   object_observations_[object_observations_.size()] = new_object_observation;
 }
 
-CameraGroupObs::~CameraGroupObs() {
-  for (auto const &item : object_pose_)
-    delete[] item.second;
-  object_pose_.clear();
-}
+CameraGroupObs::~CameraGroupObs() {}
```

Comment:
Is it safe just like that?

Reply:
Yes, because we use `std::vector` for pose and it has proper memory management with it.